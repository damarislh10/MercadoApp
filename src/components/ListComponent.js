import React, { useEffect, useState } from "react";
import { Container, Row } from "react-bootstrap";
import { url as endpoint } from "../helpers/url";
import MercadoCard from "./MercadoCard";
import "../styles/listC.css";
const ListComponent = () => {
  const [product, setProduct] = useState([]);

  useEffect(() => {
    getData();
  }, []);

  const getData = async () => {
    const resp = await fetch(endpoint);
    const { data } = await resp.json();
    setProduct(data);
  };
  return (
    <div>
      <Container>
        <Row style={{ boxShadow: "none" }} className="row-card" lg={4}>
          {product.map((pr) => (
            <MercadoCard key={pr.id} {...pr} />
          ))}
        </Row>
      </Container>
    </div>
  );
};

export default ListComponent;
