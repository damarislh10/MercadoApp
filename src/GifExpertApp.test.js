import React from 'react';
import {shallow} from 'enzyme';
import ListComponent from './components/ListComponent';

describe('Pruebas en <ListComponent/>',()=>{
    test('Debe mostrarse correctamente',()=>{

        const wrapper = shallow(<ListComponent/>)
        expect(wrapper).toMatchSnapshot()
    })
})

