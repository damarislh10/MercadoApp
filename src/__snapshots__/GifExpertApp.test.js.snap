// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ListComponent/> Debe mostrarse correctamente 1`] = `
<Container
  fluid={false}
>
  <Row
    className="row-card"
    lg={4}
    style={
      Object {
        "boxShadow": "none",
      }
    }
  />
</Container>
`;
